spec/unit/file_system_spec.rb

```diff
@@ -649,5 +649,49 @@ def increment_counter_in_multiple_processes(file, num_procs, options)
         end.to raise_error(Errno::EEXIST)
       end
     end
+
+    describe "expand_path", :if => Puppet::Util::Platform.windows? do
+      let(:nonexist_file) { 'C:\\file~1.ext' }
+      let(:nonexist_path) { 'C:\\progra~1\\missing\\path\\file.ext' }
+
+      it "should behave the same as Ruby's expand_path for a file that doesn't exist" do
+        expect(Puppet::FileSystem.exist?(nonexist_file)).to be_falsey
+        # this will change c:\\file~1.ext to c:/file~1.ext (existing Ruby behavior), but not expand any ~
+        ruby_expanded = File.expand_path(nonexist_file)
+        expect(ruby_expanded).to match(/~/)
+        expect(Puppet::FileSystem.expand_path(nonexist_file)).to eq(ruby_expanded)
+      end
+
+      it "should behave the same as Ruby's expand_path for a file with a parent path that doesn't exist" do
+        expect(Puppet::FileSystem.exist?(nonexist_path)).to be_falsey
+        # this will change c:\\progra~1 to c:/progra~1 (existing Ruby behavior), but not expand any ~
+        ruby_expanded = File.expand_path(nonexist_path)
+        expect(ruby_expanded).to match(/~/)
+        expect(Puppet::FileSystem.expand_path(nonexist_path)).to eq(ruby_expanded)
+      end
+
+      it "should expand a shortened path completely, unliked Ruby File.expand_path" do
+        tmp_long_dir = tmpdir('super-long-thing-that-Windows-shortens')
+        short_path = Puppet::Util::Windows::File.get_short_pathname(tmp_long_dir)
```

Comment:
I implemented Windows `GetShortPathName` API solely for this test.  @puppetlabs/windows any thoughts about that?